common4j/src/main/com/microsoft/identity/common/java/cache/SharedPreferencesAccountCredentialCache.java

```diff
@@ -100,6 +100,14 @@ public SharedPreferencesAccountCredentialCache(
         Logger.verbose(TAG, "Init: " + TAG);
         mSharedPreferencesFileManager = sharedPreferencesFileManager;
         mCacheValueDelegate = accountCacheValueDelegate;
+        if (mSharedPreferencesFileManager.get(SHA1_APPLICATION_IDENTIFIER_ACCESS_TOKEN_CLEARED) == null) {
+            for (final Credential credential : getCredentials()) {
+                if (isSha1ApplicationIdentifierAccessToken(credential)) {
+                    removeCredential(credential);
+                }
+            }
+            mSharedPreferencesFileManager.put(SHA1_APPLICATION_IDENTIFIER_ACCESS_TOKEN_CLEARED, String.valueOf(true));
+        }
```

Comment:
We don't get another token here, we just remove the existing one, right? So I'm assuming the new token will be acquired through a silent request after the user updates the broker-hosting app on their device?

Reply:
Yep!